docs/build/wasm/contract_environment/chain-extension/chain_extensions.md

```diff
@@ -0,0 +1,57 @@
+---
+sidebar_position: 1
+---
+
+# Chain Extensions
+
+Chain extension is a way to extend contracts API to add contracts to runtime pallet interaction. By default, contracts can only do cross-contract calls within their environment (pallet-contracts). Chain extension allows to add custom callable pallet functions.
+
+![ink-ce](../../img/ink-ce.png)
+
+### What chain extensions are available ?
+
+#### XVM
+
+This chain extension enable usage of XVM in your contracts. More info in the [ink! XVM SDK repo](https://github.com/AstarNetwork/ink-xvm-sdk).
+
+#### Dapps Staking
+
+This chain extension adds call to `pallet_dapps_staking` so that you can use Dapp Staking in your contracts. More info in the [chain-extensions contracts repo](https://github.com/swanky-dapps/chain-extension-contracts).
+
+#### Assets
+
+This chain extension adds call to `pallet_assets` so that you can use Assets in your contracts. More info in the [chain-extensions contracts repo](https://github.com/swanky-dapps/chain-extension-contracts).
+
+### Availability in networks
+
+
+| Chain extension | Swanky | Shibuya | Shiden | Astar |
+|---|---|---|---|---|
+| XVM | :white_large_square: | :white_check_mark: | :white_large_square: | :white_large_square: |
+| Dapps Staking |  :white_check_mark:| :white_check_mark: | :white_large_square: | :white_large_square: |
+| Assets | :white_large_square: | :white_large_square: | :white_large_square: | :white_large_square: |
+
+
+### Implementations
+
+There is two implementations: one in the runtime and one in ink! side.  
+
+#### Runtine
+
+Implementation of the chain extension on runtime side is available on [astar-frame repository](https://github.com/AstarNetwork/astar-frame/tree/polkadot-v0.9.37/chain-extensions)
+
+#### ink! implementation
+
+On contract side the implementation is made using [ChainExtensionMethod](https://github.com/paritytech/ink/blob/db7a906522a7e97ed5057b193df1253b33e99ee4/crates/env/src/chain_extension.rs#L77) that uses a custom environment
+(so it can be used with other libraries that use custom environment like OpenBrush). It is implementing as a crate that you can import in you contract. It can be found in [chain-extension contracts repository](https://github.com/swanky-dapps/chain-extension-contracts)
```

Comment:
```suggestion
(so it can be used with other libraries that use custom environment like OpenBrush). It is implemented as a crate that you can import in you contract. It can be found in [chain-extension contracts repository](https://github.com/swanky-dapps/chain-extension-contracts)
```